src/ompl/geometric/planners/rrt/STRRTstar.h

```diff
@@ -0,0 +1,547 @@
+                    OMPL_ERROR("%s: Time Bound Factor Increase needs to be higher than 1.", getName().c_str());
+                }
+                timeBoundFactorIncrease_ = f;
+            }
+
+            void setInitialTimeBoundFactor(double f)
+            {
+                if (f <= 1.0)
+                {
+                    OMPL_ERROR("%s: Initial Time Bound Factor Increase needs to be higher than 1.", getName().c_str());
+                }
+                initialTimeBoundFactor_ = f;
+            }
+
+            void setSampleUniformForUnboundedTime(bool uniform)
+            {
+                sampleUniformForUnboundedTime_ = uniform;
+            }
+
+            void setup() override;
+
+        protected:
+            /** \brief Representation of a motion */
+            class Motion
+            {
+            public:
+                Motion() = default;
+
+                explicit Motion(const base::SpaceInformationPtr &si) : state(si->allocState())
+                {
+                }
+
+                ~Motion() = default;
+
+                const base::State *root{nullptr};
+                base::State *state{nullptr};
+                Motion *parent{nullptr};
+                /** \brief The set of motions descending from the current motion */
+                std::vector<Motion *> children{};
+                // only used by goal tree
+                Motion *connectionPoint{nullptr};  // the start tree motion, if there is a direct connection
+                int numConnections{0};  // number of connections to the start tree of self and all descendants
+            };
+
+            /** \brief Optimization Objective to minimize time at which any goal can be achieved.
+             * It is not possible to set a different optimization objective. */
+            class MinimizeGoalTime : public base::OptimizationObjective
+            {
+            public:
+                explicit MinimizeGoalTime(const base::SpaceInformationPtr &si) : OptimizationObjective(si)
+                {
+                }
+
+            private:
+                base::Cost stateCost(const ompl::base::State *s) const override
+                {
+                    return base::Cost(s->as<base::CompoundState>()->as<base::TimeStateSpace::StateType>(1)->position);
+                }
+
+                base::Cost motionCost(const ompl::base::State *s1, const ompl::base::State *s2) const override
+                {
+                    return combineCosts(stateCost(s1), stateCost(s2));
+                }
+
+                base::Cost combineCosts(ompl::base::Cost c1, ompl::base::Cost c2) const override
+                {
+                    return c1.value() > c2.value() ? c1 : c2;
+                }
+
+                base::Cost identityCost() const override
+                {
+                    return base::Cost(-std::numeric_limits<double>::infinity());
+                }
+            };
```

Comment:
I think you better put this into a new class. You can then put this under base/objectives.